crates/router_env/src/request_id.rs

```diff
@@ -0,0 +1,365 @@
+#![deny(unused, missing_docs)]
+//! Request ID middleware for Actix Web applications.
+//!
+//! This module provides middleware to associate every HTTP request with a unique identifier.
+//! The request ID can be used for distributed tracing, error tracking, and request correlation
+//! across microservices.
+//!
+//! # Features
+//!
+//! - **Automatic UUID v7 generation**: Uses time-ordered UUIDs for better performance
+//! - **Header reuse**: Respects incoming request ID headers for request correlation
+//! - **Configurable headers**: Custom header names supported for any request ID header
+//! - **Request extraction**: RequestId can be extracted in route handlers
+//! - **Response headers**: Automatically adds request ID to response headers
+//! - **Upstream logging**: Logs incoming request IDs for debugging
+//! - **Exact parity**: Supports `.as_hyphenated()` method like tracing_actix_web::RequestId
+//!
+//! # Examples
+//!
+//! ```rust
+//! use router_env::{RequestIdentifier, IdReuse};
+//! use actix_web::{web, App, HttpServer};
+//!
+//! let app = App::new()
+//!     .wrap(
+//!         RequestIdentifier::new("x-request-id")
+//!             .use_incoming_id(IdReuse::UseIncoming)
+//!     );
+//! ```
+
+use std::{
+    fmt::{self, Display, Formatter},
+    future::{ready, Future, Ready},
+    pin::Pin,
+    str::FromStr,
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+use actix_web::{
+    dev::{Payload, Service, ServiceRequest, ServiceResponse, Transform},
+    error::ResponseError,
+    http::header::{HeaderName, HeaderValue},
+    Error as ActixError, FromRequest, HttpMessage, HttpRequest,
+};
+use uuid::Uuid;
+
+/// Custom result type for request ID operations.
+pub type RequestIdResult<T> = Result<T, Error>;
+
+/// Errors that can occur when working with request IDs.
+#[derive(Debug, Clone)]
+pub enum Error {
```

Comment:
Please name this better, since there is a trait with the same name from the standard library.